From 91854b1db39e29db1d6668fbb807b1f7c0a09c2b Mon Sep 17 00:00:00 2001
From: Adriaan de Groot <groot@kde.org>
Date: Sat, 9 Mar 2019 13:59:00 -0500
Subject: [PATCH] (FreeBSD) Link to libutil

This is preparatory work: the sddm-helper is going to need
libutil for setusercontext(), which handles integration with
login.conf and per-user configurations (e.g. language settings,
CPU affinity and resource limits).
---
 src/helper/CMakeLists.txt | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/helper/CMakeLists.txt b/src/helper/CMakeLists.txt
index ebf4a6e..9fe9278 100644
--- a/src/helper/CMakeLists.txt
+++ b/src/helper/CMakeLists.txt
@@ -1,3 +1,5 @@
+include(CheckLibraryExists)
+
 include_directories(
     "${CMAKE_SOURCE_DIR}/src/common"
     "${CMAKE_SOURCE_DIR}/src/auth"
@@ -32,6 +34,18 @@ endif()
 
 add_executable(sddm-helper ${HELPER_SOURCES})
 target_link_libraries(sddm-helper Qt5::Network Qt5::DBus Qt5::Qml)
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+    # On FreeBSD (possibly other BSDs as well), we want to use
+    # setusercontext() to set up the login configuration from login.conf
+    find_library(_have_libutil util)
+    if(_have_libutil)
+        check_library_exists(${_have_libutil} setusercontext "" _have_setusercontext)
+    endif()
+    if(_have_libutil AND _have_setusercontext)
+        target_link_libraries(sddm-helper ${_have_libutil})
+    endif()
+endif()
+
 if(PAM_FOUND)
     target_link_libraries(sddm-helper ${PAM_LIBRARIES})
 else()
-- 
2.23.0

